recipe/meta.yaml

```diff
@@ -37,13 +41,45 @@ requirements:
     - h2 >=3,<5
     - socksio >=1,<2
 
+{% set tests_to_skip = "" %}
+# skip tests using socksio, optional dependency, not available for all platforms
+{% set tests_to_skip = tests_to_skip + " test_socks_proxy_deprecated" %}               
+{% set tests_to_skip = tests_to_skip + " or test_socks_proxy" %}
+# skip some tests for Windows
+#
+# no TypeError raised
+{% set tests_to_skip = tests_to_skip + " or test_multipart_encode_files_raises_exception_with_text_mode_file" %}  # [win]
+# no httpx.WriteTimeout
+{% set tests_to_skip = tests_to_skip + " or test_write_timeout[asyncio]" %}            # [win]
+# paths with "\\\\" instead of "\\"
+{% set tests_to_skip = tests_to_skip + " or test_logging_ssl" %}                       # [win]
+# "tests\\test_utils.py" != "tests/test_utils.py"
+{% set tests_to_skip = tests_to_skip + " or test_get_ssl_cert_file" %}                 # [win]
+
 test:
+  source_files:
+    - tests
   imports:
     - httpx
   requires:
     - pip
+    - pytest
+    # You want brotli-python here rather than just the brotli that
+    # "import brotli" might imply
+    - brotli-python
+    - chardet
+    - trio
+    - trustme
+    - uvicorn
+    - h2
+    # [cli testing] - pygment unavailable but optional
+    - click
+    # - pygment
+    - rich
   commands:
     - pip check
+    - httpx --help
```

Comment:
this fails on `win` at the moment with:
```
Traceback (most recent call last):
  File "C:\Users\lorepirri\miniconda3\envs\c3i\conda-bld\httpx_1723052686727\_test_env\Scripts\httpx-script.py", line 10, in <module>
    sys.exit(main())
  File "C:\Users\lorepirri\miniconda3\envs\c3i\conda-bld\httpx_1723052686727\_test_env\lib\site-packages\click\core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "C:\Users\lorepirri\miniconda3\envs\c3i\conda-bld\httpx_1723052686727\_test_env\lib\site-packages\click\core.py", line 1077, in main
    with self.make_context(prog_name, args, **extra) as ctx:
  File "C:\Users\lorepirri\miniconda3\envs\c3i\conda-bld\httpx_1723052686727\_test_env\lib\site-packages\click\core.py", line 943, in make_context
    self.parse_args(ctx, args)
  File "C:\Users\lorepirri\miniconda3\envs\c3i\conda-bld\httpx_1723052686727\_test_env\lib\site-packages\click\core.py", line 1408, in parse_args
    value, args = param.handle_parse_result(ctx, opts, args)
  File "C:\Users\lorepirri\miniconda3\envs\c3i\conda-bld\httpx_1723052686727\_test_env\lib\site-packages\click\core.py", line 2400, in handle_parse_result
    value = self.process_value(ctx, value)
  File "C:\Users\lorepirri\miniconda3\envs\c3i\conda-bld\httpx_1723052686727\_test_env\lib\site-packages\click\core.py", line 2362, in process_value
    value = self.callback(ctx, self, value)
  File "C:\Users\lorepirri\miniconda3\envs\c3i\conda-bld\httpx_1723052686727\_test_env\lib\site-packages\httpx\_main.py", line 307, in handle_help
    print_help()
  File "C:\Users\lorepirri\miniconda3\envs\c3i\conda-bld\httpx_1723052686727\_test_env\lib\site-packages\httpx\_main.py", line 27, in print_help
    console.print("[bold]HTTPX :butterfly:", justify="center")
  File "C:\Users\lorepirri\miniconda3\envs\c3i\conda-bld\httpx_1723052686727\_test_env\lib\site-packages\rich\console.py", line 1721, in print
    self._buffer.extend(new_segments)
  File "C:\Users\lorepirri\miniconda3\envs\c3i\conda-bld\httpx_1723052686727\_test_env\lib\site-packages\rich\console.py", line 865, in __exit__
    self._exit_buffer()
  File "C:\Users\lorepirri\miniconda3\envs\c3i\conda-bld\httpx_1723052686727\_test_env\lib\site-packages\rich\console.py", line 823, in _exit_buffer
    self._check_buffer()
  File "C:\Users\lorepirri\miniconda3\envs\c3i\conda-bld\httpx_1723052686727\_test_env\lib\site-packages\rich\console.py", line 2027, in _check_buffer
    legacy_windows_render(buffer, LegacyWindowsTerm(self.file))
  File "C:\Users\lorepirri\miniconda3\envs\c3i\conda-bld\httpx_1723052686727\_test_env\lib\site-packages\rich\_windows_renderer.py", line 17, in legacy_windows_render
    term.write_styled(text, style)
  File "C:\Users\lorepirri\miniconda3\envs\c3i\conda-bld\httpx_1723052686727\_test_env\lib\site-packages\rich\_win32_console.py", line 442, in write_styled
    self.write_text(text)
  File "C:\Users\lorepirri\miniconda3\envs\c3i\conda-bld\httpx_1723052686727\_test_env\lib\site-packages\rich\_win32_console.py", line 403, in write_text
    self.write(text)
  File "C:\Users\lorepirri\miniconda3\envs\c3i\conda-bld\httpx_1723052686727\_test_env\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\U0001f98b' in position 6: character maps to <undefined>
WARNING: Tests failed for httpx-0.27.0-py39haa95532_0.tar.bz2 - moving package to C:\Users\lorepirri\miniconda3\envs\c3i\conda-bld\broken
TESTS FAILED: httpx-0.27.0-py39haa95532_0.tar.bz2
```

Reply:
on linux:

<img width="1554" alt="image" src="https://github.com/user-attachments/assets/415312d0-ecae-408a-b3e0-907e7b4f29ab">

Reply:
```
                                   HTTPX ðŸ¦‹                                    

                        A next generation HTTP client.
```

On a windows dev box I'm playing with

Reply:
Oh, and a:

```
______________________________ test_get[asyncio] ______________________________

server = <tests.conftest.TestServer object at 0x00000271CC3BE2D0>

    @pytest.mark.anyio
    async def test_get(server):
        url = server.url
        async with httpx.AsyncClient(http2=True) as client:
            response = await client.get(url)
        assert response.status_code == 200
        assert response.text == "Hello, world!"
        assert response.http_version == "HTTP/1.1"
        assert response.headers
        assert repr(response) == "<Response [200 OK]>"
>       assert response.elapsed > timedelta(seconds=0)
E       assert datetime.timedelta(0) > datetime.timedelta(0)
E        +  where datetime.timedelta(0) = <Response [200 OK]>.elapsed
E        +  and   datetime.timedelta(0) = timedelta(seconds=0)

tests\client\test_async_client.py:21: AssertionError
---------------------------- Captured stdout call -----------------------------
INFO:     127.0.0.1:50226 - "GET / HTTP/1.1" 200 OK
```

🤷

Reply:
`set PYTHONIOENCODING=utf8  # [win]` solves this during tests. Running the command from a Windows terminal works.

Reply:
It should work on terminals with proper fonts:

<img width="1586" alt="image" src="https://github.com/user-attachments/assets/a7fd7606-fd87-4b1a-8a9b-7fbd0a313f26">

Reply:
<img width="1179" alt="image" src="https://github.com/user-attachments/assets/967145bb-ab31-47be-8911-5971c62a5ac5">

Reply:
Those last two are from RDP?

Reply:
RDP as in the remote desktop? No, they are from a Parallels VM.

Reply:
OK, yeah, "on the console" then and using native utilities, rather than through a some terminal session which may disagree about protocols.

Reply:
It does not crash, and the `cli` is optional.